cli/tests/pcluster/validators/test_iam_validators.py

```diff
@@ -113,3 +115,125 @@ def test_additional_iam_policy_validator(mocker, policy_arn, expected_get_policy
     mocker.patch("pcluster.aws.iam.IamClient.get_policy", side_effect=expected_get_policy_side_effect)
     actual_failures = AdditionalIamPolicyValidator().execute(policy=policy_arn)
     assert_failure_messages(actual_failures, expected_message)
+
+
+@pytest.mark.parametrize(
+    "resource_prefix, expected_message,expected_failure_level",
+    [
+        (
+            r"\pathprefix\\",
+            "Unsupported format for ResourcePrefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "",
+            "Unsupported format for ResourcePrefix ",
+            FailureLevel.ERROR,
+        ),
+        (
+            "/",
+            "Unsupported format for ResourcePrefix /",
+            FailureLevel.ERROR,
+        ),
+        (
+            "//",
+            "Unsupported format for ResourcePrefix //",
+            FailureLevel.ERROR,
+        ),
+        (
+            "///////",
+            "Unsupported format for ResourcePrefix ///////",
+            FailureLevel.ERROR,
+        ),
+        (
+            "/prefix",  # This is not pathprefix ---- Need to check
+            "Unsupported format for ResourcePrefix /prefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            ",./pathprefix/",
+            "Unsupported format for ResourcePrefix ,./pathprefix/",
+            FailureLevel.ERROR,
+        ),
+        (
+            ";)*/anything/pathprefix/",
+            "Unsupported format for ResourcePrefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "#$/pathprefix/anything/",
+            "Unsupported format for ResourcePrefix ",
+            FailureLevel.ERROR,
+        ),
```

Comment:
We can collapse these cases into a single one

Reply:
Sure